install_latest_blue_onyx.ps1

```diff
@@ -48,85 +48,103 @@ try {
-    
-    Write-Host "Version: $($json.version)"
-    Write-Host "ZIP URL: $zipUrl"
-    Write-Host "SHA256 URL: $sha256Url"
-    
+
+    Write-Host "Version: $($json.version)" -ForegroundColor Cyan
+    Write-Host "ZIP URL: $zipUrl" -ForegroundColor Blue
+    Write-Host "SHA256 URL: $sha256Url" -ForegroundColor Yellow
+
     # --- 4. Check if ZIP and SHA256 already exist in %TEMP%, else download (cache) ---
     $zipFile    = Join-Path $tempPath $json.windows
     $sha256File = Join-Path $tempPath $json.windows_sha256
-    
+
     if (Test-Path $zipFile) {
-        Write-Host "Found cached ZIP file at $zipFile. Skipping download..."
+        Write-Host "Found cached ZIP file at $zipFile. Skipping download..." -ForegroundColor Green
     }
     else {
-        Write-Host "Downloading ZIP file to $zipFile..."
+        Write-Host "Downloading ZIP file to $zipFile..." -ForegroundColor Yellow
         Invoke-WebRequest -Uri $zipUrl -OutFile $zipFile -UseBasicParsing -ErrorAction Stop
     }
-    
+
     if (Test-Path $sha256File) {
-        Write-Host "Found cached SHA256 file at $sha256File. Skipping download..."
+        Write-Host "Found cached SHA256 file at $sha256File. Skipping download..." -ForegroundColor Green
     }
     else {
-        Write-Host "Downloading SHA256 file to $sha256File..."
+        Write-Host "Downloading SHA256 file to $sha256File..." -ForegroundColor Green
         Invoke-WebRequest -Uri $sha256Url -OutFile $sha256File -UseBasicParsing -ErrorAction Stop
     }
-    
+
     # --- 5. Verify the ZIP file's integrity with the SHA256 ---
-    Write-Host "Verifying ZIP file integrity..."
+    Write-Host "Verifying ZIP file integrity..." -ForegroundColor DarkYellow
     $sha256FileContent = Get-Content $sha256File -Raw
     $pattern = '[A-Fa-f0-9]{64}'  # 64 hex characters for a SHA256 hash
-    
+
     $match = [System.Text.RegularExpressions.Regex]::Match($sha256FileContent, $pattern)
     if (-not $match.Success) {
         throw "Could not parse a valid 64-hex SHA256 from the .sha256 file."
     }
     $expectedSha256 = $match.Value.ToLower()
-    
+
     $actualHash = (Get-FileHash -Algorithm SHA256 $zipFile).Hash.ToLower()
-    
-    Write-Host "Expected SHA256: $expectedSha256"
-    Write-Host "Actual   SHA256: $actualHash"
-    
+
+    Write-Host "Expected SHA256: $expectedSha256" -ForegroundColor Yellow
+    Write-Host "Actual   SHA256: $actualHash" -ForegroundColor Yellow
+
     if ($expectedSha256 -ne $actualHash) {
         throw "ZIP file SHA256 does not match expected value!"
     }
-    Write-Green "SHA256 verification successful."
-    
+    Write-Host "SHA256 verification successful." -ForegroundColor Green
+
     # --- 6. Extract the ZIP into a temporary subfolder, then flatten if needed ---
-    $destinationPath = Join-Path $env:USERPROFILE ".blue-onyx"
+
+    # Get the desired folder from the user using a file dialog
```

Comment:
I think this popup is fine, but I think we should allow to either read an environment variable or have the abillity to pass in the path so one can automate this as a task. Think run this script once per week as task and it will check for updates, update and restart the process

Reply:
i think we might start dipping into MSI installer and registry writing territory to write "where" things are installed and what versions are installed. I dont have familiarity into doing any of that but i can google my ass off. But also you might need an agent file that does the parodical checking, maybe writing to a windows task schedule to do that.

Reply:
Yeah it is a fine line. I was hoping to be able to avoid installers. I was looking into inno installer before I went down the powershell script path.

Feel free to punt this for now.

Reply:
Testing a popup box option.